internal/errors/http/echo/default.go

```diff
@@ -0,0 +1,62 @@
+// package echo
+//
+// This package implements the default error handler to be
+// used for the api service to match the openapi specification.
+//
+// See: https://echo.labstack.com/docs/error-handling#custom-http-error-handler
+package echo
+
+import (
+	"errors"
+	"log/slog"
+	"net/http"
+	"strconv"
+
+	"github.com/labstack/echo/v4"
+	"github.com/openlyinc/pointy"
+	api_public "github.com/podengo-project/idmsvc-backend/internal/api/public"
+	app_context "github.com/podengo-project/idmsvc-backend/internal/infrastructure/context"
+)
+
+func DefaultErrorHandler(err error, c echo.Context) {
+	logger := app_context.LogFromCtx(c.Request().Context())
+	errList := make([]api_public.ErrorInfo, 0, 10)
+
+	firstCode := 0
+	response := api_public.ErrorResponse{}
+	for {
+		logger.Debug("encapsulating error", slog.String("msg", err.Error()))
+		errInfo, code := parseError(err)
+		errList = append(errList, errInfo)
+		if firstCode == 0 {
+			firstCode = code
+		}
+		if err = errors.Unwrap(err); err == nil {
+			break
+		}
+	}
+	response.Errors = &errList
+	if err = c.JSON(firstCode, &response); err != nil {
+		logger.Error("the error could not be reported in the http response",
+			slog.String("msg", err.Error()),
+		)
+	}
+}
+
+// parseError translate an error into api_public.ErrorInfo and
+// return the first http status code of the error list.
+func parseError(err error) (api_public.ErrorInfo, int) {
+	switch value := err.(type) { //nolint:all
+	case *echo.HTTPError:
+		return api_public.ErrorInfo{
+			Status: strconv.Itoa(value.Code),
+			Title:  value.Error(), /* TODO ?? // value.Message, */
```

Comment:
I'd use `Message` as `Code` is put to `Status` and would be redundant in `Title`.  
The implementation of `.Error()` is:
```
func (he *HTTPError) Error() string {
	if he.Internal == nil {
		return fmt.Sprintf("code=%d, message=%v", he.Code, he.Message)
	}
	return fmt.Sprintf("code=%d, message=%v, internal=%v", he.Code, he.Message, he.Internal)
}
```
`Internal`, I believe, is unwrapped as another error.